llvm/lib/Target/X86/X86ISelLowering.cpp

```diff
@@ -47428,6 +47428,8 @@ static SDValue combineShiftRightArithmetic(SDNode *N, SelectionDAG &DAG,
   APInt SarConst = N1->getAsAPIntVal();
   EVT CVT = N1.getValueType();
 
+  if (CVT != N01.getValueType())
```

Comment:
Can you fix this comment on line 47412 `depending on sign of (SarConst - [56,48,32,24,16])`

Reply:
Can you also fix 47411 to say `ashr` instead of `lshr` to match what the code does

Reply:
I've cleaned up a bit now:
- fixed the lshr->ashr thing (actually using SRA/SHL now, instead of the IR names when describing the folds)
- renamed SarConst -> SraConst
- got rid of the [56,48,32,24,16] comments (I did not really understand those comments and they did not fully match what the code was doing afaict)